core-api/src/test/resources/config/valid-project-config-v4.json

```diff
@@ -27,6 +27,43 @@
       "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"nationality\", \"type\": \"custom_attribute\", \"value\": \"English\"}, {\"name\": \"nationality\", \"type\": \"custom_attribute\"}]]]"
     }
   ],
+  "typedAudiences": [
+    {
+      "id": "3468206643",
+      "name": "BOOL",
+      "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"booleanKey\", \"type\": \"custom_attribute\", \"match\":\"exact\", \"value\":true}]]]"
+    },
+    {
+      "id": "3468206644",
+      "name": "INT",
+      "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"integerKey\", \"type\": \"custom_attribute\", \"match\":\"gt\", \"value\":1.0}]]]"
+    },
+    {
+      "id": "3468206645",
+      "name": "DOUBLE",
+      "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"doubleKey\", \"type\": \"custom_attribute\", \"match\":\"lt\", \"value\":100.0}]]]"
+    },
+    {
+      "id": "3468206642",
+      "name": "Gryffindors",
+      "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"house\", \"type\": \"custom_attribute\", \"match\":\"exact\", \"value\":\"Gryffindor\"}]]]"
+    },
+    {
+      "id": "3988293898",
+      "name": "Slytherins",
+      "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"house\", \"type\": \"custom_attribute\", \"match\":\"substring\", \"value\":\"Slytherin\"}]]]"
+    },
+    {
+      "id": "4194404272",
+      "name": "english_citizens",
+      "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"nationality\", \"type\": \"custom_attribute\", \"match\":\"exact\", \"value\":\"English\"}]]]"
+    },
+    {
+      "id": "2196265320",
+      "name": "audience_with_missing_value",
+      "conditions": "[\"and\", [\"or\", [\"or\", {\"name\": \"nationality\", \"type\": \"custom_attribute\", \"value\": \"English\"}, {\"name\": \"nationality\", \"type\": \"custom_attribute\"}]]]"
+    }
+  ],
```

Comment:
Since you'd asked me to check: this looks accurate. But I'd also test with an audience whose ID is found in both `audiences` and `typedAudiences`.

It might also be wise to include audience conditions with `match` being `null` instead of being undefined, though I have no _intention_ of having the app backend serialize such conditions. 🤷

Reply:
All the string conditions that map to exact are also in the audience list and replace the audience lookup map with the typed audience since they share ids.  Look at the audience list above typed audience definition.

Reply:
Ah, gotcha. I should've expanded the whole file and checked 😬